pkg/liqo-controller-manager/namespace-controller/namespace_controller.go

```diff
@@ -1,147 +1,86 @@
-/*
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-   http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
 package namespacectrl
 
 import (
 	"context"
 
-	mapsv1alpha1 "github.com/liqotech/liqo/apis/virtualKubelet/v1alpha1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	offv1alpha1 "github.com/liqotech/liqo/apis/offloading/v1alpha1"
 	liqoconst "github.com/liqotech/liqo/pkg/consts"
 
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/klog"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	ctrlutils "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
-// NamespaceReconciler changes DesiredMapping field of NamespaceMaps if a remote Namespace must be created or deleted.
+// NamespaceReconciler checks presence of enabling Liqo label on the Namespace.
 type NamespaceReconciler struct {
 	client.Client
 	Scheme *runtime.Scheme
 }
 
-const (
-	mappingLabel                 = "mapping.liqo.io"
-	offloadingLabel              = "offloading.liqo.io"
-	offloadingPrefixLabel        = "offloading.liqo.io/"
-	namespaceControllerFinalizer = "namespace-controller.liqo.io/finalizer"
-)
-
-// Reconcile checks presence of right labels on the Namespace.
+// Reconcile checks presence of enabling Liqo label on the Namespace.
 func (r *NamespaceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	namespace := &corev1.Namespace{}
-	if err := r.Get(context.TODO(), req.NamespacedName, namespace); err != nil {
+	if err := r.Get(ctx, req.NamespacedName, namespace); err != nil {
 		klog.Errorf("%s --> Unable to get namespace '%s'", err, req.Name)
```

Comment:
I am not sure this is always an error. What about a deleted element?

Reply:
You are right, thank you